frost-ristretto255/src/frost.rs

```diff
@@ -655,91 +918,25 @@ pub fn aggregate(
     );
 
     for signing_share in signing_shares {
-        let signer_pubkey = pubkeys.signer_pubkeys[&signing_share.index];
+        let signer_pubkey = pubkeys.signer_pubkeys.get(&signing_share.index).unwrap();
         let lambda_i = generate_lagrange_coeff(signing_share.index, signing_package)?;
-        let signer_commitment = signing_package
-            .signing_commitments
-            .iter()
-            .find(|comm| comm.index == signing_share.index)
-            .ok_or("No matching signing commitment for signer")?;
-
-        let commitment_i = signer_commitment.hiding.0
-            + (signer_commitment.binding.0 * bindings[&signing_share.index]);
 
-        signing_share.check_is_valid(&signer_pubkey, lambda_i, commitment_i, challenge)?;
+        signing_share.verify(signer_pubkey, lambda_i, challenge)?;
     }
 
     // The aggregation of the signature shares by summing them up, resulting in
     // a plain Schnorr signature.
     let mut z = Scalar::zero();
+    let mut R: RistrettoPoint = RistrettoPoint::identity();
+
     for signature_share in signing_shares {
-        z += signature_share.signature.0;
+        z += signature_share.signature.z_share;
+        // TODO(dconnolly): enforce this == group_commitment? This is not in the spec.
+        R += signature_share.signature.R_share;
```

Comment:
When deriving the commitment, we should be using values sent strictly in round one. Otherwise, the signer can substitute whatever they want in round two.

Reply:
Updated